wewatch-client/src/components/Lobby/Lobby.js

```diff
@@ -78,40 +80,55 @@ const Lobby = ({location}) => {
+        socket.on('sessionMembers', ({roomId, users, host}) => {
             // TODO do something with the returned data
-            console.log(top10)
-            setGoSwipe(true);
+            setPage('Swiping');
         });
     }, []);
 
-    return (
-        <div className="outerContainer">
-            <div className="container">
-                <h1>Welcome to {hostName}'s Lobby</h1>
-                <h2>Lobby ID: {roomId} </h2>
-                <h2>In the lobby:</h2>
-                <ul>
-                    {users.map(element => {
-                        if(element.name === name) {
-                            return (<li>{element.name + " (You)"}</li>)
-                        } else if (element.name === hostName) {
-                            return (<li>{element.name + " (Host)"}</li>)
-                        } else {
-                            return (<li>{element.name}</li>)
-                        }
-                    })}
-                </ul>
-                {name === hostName 
-                    ? 
-                    <button className={'button mt-20'} type="button" onClick={onClickStartSession}>Start Session</button> 
-                    :
-                    <h2>Waiting for Host to start!</h2> 
-                }
-                { goSwipe ? <Redirect to='/swiping?room=${roomId}'/> : null }
+    useEffect(() => {
+        socket.on('noMatchRedirect', () => {
+            setPage('NoMatch')
+      });
+    }, []);
+
+
+    //MAIN PAGES
+    function LobbyPage(props) {
+        return (
+            <div className="outerContainer">
+                <div className="container">
+                    <h1>Welcome to {hostName}'s Lobby</h1>
+                    <h2>Lobby ID: {roomId} </h2>
+                    <h2>In the lobby:</h2>
+                    <ul>
+                        {users.map(element => {
+                            if(element.name === name) {
+                                return (<li>{element.name + " (You)"}</li>)
+                            } else if (element.name === hostName) {
+                                return (<li>{element.name + " (Host)"}</li>)
+                            } else {
+                                return (<li>{element.name}</li>)
+                            }
+                        })}
+                    </ul>
+                    {name === hostName 
+                        ? 
+                        <button className={'button mt-20'} type="button" onClick={onClickStartSession}>Start Session</button> 
+                        :
+                        <h2>Waiting for Host to start!</h2> 
+                    }
+                </div>
             </div>
-        </div>
-    );
-  }
+        ); 
+    }
+
+    if (page == "Lobby"){
+        return <LobbyPage/>;
+    } else if (page == "Swiping") {
+        return (<div><button onClick={nomatchtest}>No Match test (will be removed)</button><Swiping/></div>);
```

Comment:
Are we keeping the button here or are we going to render another component? Confused here because of the message in the parentheses. Let me know your thoughts.